posix/unix.c

```diff
@@ -997,21 +1037,24 @@ int unix_poll(unsigned socket, short events)
 	else {
 		if (events & (POLLIN | POLLRDNORM | POLLRDBAND)) {
 			proc_lockSet(&s->lock);
-			if (_cbuffer_avail(&s->buffer) > 0)
+			if (_cbuffer_avail(&s->buffer) > 0 || (s->connect != NULL && (s->state & US_LISTENING))) {
```

Comment:
MISRA guideline: use `(s->state & US_LISTENING) != 0` style here and elsewhere. In general control expressions should have Boolean type.